src/config/kernel.md

```diff
@@ -46,6 +46,61 @@ want to remove the default kernel packages, use
 you will be able to remove the packages with
 [xbps-remove(1)](https://man.voidlinux.org/xbps-remove.1).
 
+## cmdline
+
+The kernel, the initial RAM disk (initrd) and some system programs can be
+configured at boot by kernel command line arguments. The parameters understood
+by the kernel are explained in the [kernel-parameters
+documentation](https://www.kernel.org/doc/html/latest/admin-guide/kernel-parameters.html)
+and by [bootparam(7)](https://man.voidlinux.org/bootparam.7). Parameters
+understood by dracut can be found in
+[dracut.cmdline(7)](https://man.voidlinux.org/dracut.cmdline.7).
+
+Once the system is booted, the current kernel command line parameters can be
+found in the `/proc/cmdline` file. Some system programs can change their
+behavior based on the parameters passed in the command line, which is what
+happens when [booting a different
+runsvdir](./services/index.md#booting-a-different-runsvdir), for example.
+
+There are different ways of setting these parameters, some of which are
+explained below.
+
+### GRUB
+
+Kernel command line arguments can be added through the GRUB bootloader by
+editing `/etc/default/grub`, changing the `GRUB_CMDLINE_LINUX_DEFAULT` variable
+and then running `update-grub`.
+
+### dracut
+
+Dracut can be configured to add additional command line arguments to the kernel
+through a configuration file. The documentation for dracut's configuration files
+can be found in [dracut.conf(5)](https://man.voidlinux.org/dracut.conf.5). To
+apply these changes, it is necessary to [regenerate](#kernel-hooks) the
+initramfs.
+
+## Kernel hardening
+
+Void Linux ships with some kernel hardening enabled by default. This was
+originally provided by kernel command line options `slub_debug=P page_poison=1`,
+but since kernel 5.3 these have been replaced with the options `init_on_alloc`
+and `init_on_free`.
+
+Void's 5.4 and newer kernels come with `init_on_alloc` pre-enabled. You should
+generally always keep that on, as it has minimal impact on performance. The
+`init_on_free` option is more expensive (~5% performance impact on average)
+and needs to be enabled explicitly with `init_on_free=1`. You can also disable
+`init_on_alloc` with `init_on_alloc=0`.
+
+There is a chance that your existing system has the old options enabled. If
+they are enabled, you should remove them, unless using a pre-5.4 kernel; on
+pre-5.4 kernels you can keep them on (or even add them) if you don't mind
+worse performance (it can be significant on older hardware).
```

Comment:
The logic seems a bit complicated to follow, I tried to simplify it a bit.

```suggestion
There is a chance that your existing system has the old options enabled. If
they are enabled, and you are using a kernel newer than `linux5.3` (`linux5.4` and on),
you should remove them. For kernels before `linux5.3` (basically the `linux4.x` packages),
 you can keep them on (or even add them) if you don't mind
worse performance (it can be significant on older hardware).
```